src/Haskell/Ide/Engine/Support/HieExtras.hs

```diff
@@ -799,3 +801,12 @@ prefixes =
   , "$c"
   , "$m"
   ]
+
+-- ---------------------------------------------------------------------
+
+getFormattingPlugin :: Config -> IdePlugins -> Maybe (PluginDescriptor, FormattingProvider)
+getFormattingPlugin config plugins = do 
```

Comment:
Can this happen inside `R` so `Config` and `IdePlugins` don’t need to be passed?

Reply:
Hardly, since IdeGhcM does not implement R, or can execute R, so other plugins would have no access to it.

Reply:
I see, I didn't notice it was being called elsewhere outside of LspStdio.hs. I feel like this should be moved into PluginIdeMonads.hs though, and made polymorphic over `MonadIde`, which should have access to both. We'll need to figure out how to give `R` access to it though

Reply:
I think what we really need is some kind of way to let plugins call each other. A bunch of plugins need to format after they do something.
Maybe a separate issue for discussion would be nice?